sql_scripts/report_digitalobjecttypes.sql

```diff
@@ -0,0 +1,17 @@
+-- This query counts all the digital objects with digital object types per each repository. 
+-- It also includes a count of all digital objects without a digital object type per repository.
+
+SELECT
+	repo.name as repository, ifnull(ev.value, "NO TYPE") as digital_object_type, COUNT(ifnull(ev.value, 0)) as count
```

Comment:
Good use of "if null"!  I've only ever used COALESCE before, but it looks like those functions are essentially the same.  Still, I'd personally switch to COALESCE, but in this case is dots not matter.

Reply:
But the real head scratcher is the count(ifnull(ev.value,0)) bit.  So, COUNT(0) is 1.  Ditto  COUNT(1), COUNT(2), etc.

Reply:
There's also the issue of the conceptual Global repo in ASpace, which can never have a digital object.  Because of all that, I'd recommend writing the script like so:

Reply:
```
-- This query counts all the digital objects with digital object types per each repository. 
-- It also includes a count of all digital objects without a digital object type per repository.
-- Now with proper 0s, due to count of digital object ids, rather than rows

SELECT
	repo.name as repository
	, COALESCE(ev.value, "NO TYPE") as digital_object_type
	, COUNT(dobj.id) AS count
FROM
	digital_object AS dobj
LEFT JOIN
	enumeration_value AS ev
    ON
    ev.id = dobj.digital_object_type_id
RIGHT JOIN
	repository AS repo
    ON
    repo.id = dobj.repo_id
WHERE 
	repo.name <> 'Global repository'
GROUP BY
	ev.value, repo.name
ORDER BY 
	repo.name ASC;

```

Reply:
Made changes to the match the above. The only one I modified was instead of using <> for removing the 'Global repository', I changed it to != so it makes more sense as I read it. I didn't know <> was a thing, thanks!